common/src/main/scala/com/gu/media/upload/Upload.scala

```diff
@@ -0,0 +1,34 @@
+package com.gu.media.upload
+
+import org.cvogt.play.json.Jsonx
+import play.api.libs.json.Format
+
+case class Upload(id: String, atomId: String, user: String, bucket: String, region: String, parts: List[UploadPart])
+
+object Upload {
+  implicit val format: Format[Upload] = Jsonx.formatCaseClass[Upload]
+
+  def calculateChunks(size: Long): List[(Long, Long)] = {
```

Comment:
I've gone one better and added a wizzy ScalaCheck properties-based test. I also refactored the code to hopefully be more readable and less `while` loopy